src/NHibernate/Proxy/IEntityNotFoundDelegate.cs

```diff
@@ -26,6 +26,14 @@ internal static class EntityNotFoundDelegateExtension
 		}
 	}
 
+	/// <summary> 
+	/// Delegate to handle the scenario of an entity not found by a specified id and property. 
+	/// </summary>
+	public interface IEntityNotFoundPropertyDelegate
```

Comment:
Can we have another name please? This is slightly confusing.

Reply:
So it's about https://github.com/nhibernate/nhibernate-core/pull/2049

It's a temp interface. So please add comment to remove it in 6.0. Something like `//6.0 TODO: Remove`

Also I think it can be considered as regression so can be retargeted to 5.3.x branch.

>Can we have another name please?

`IEntityNotFoundByUniqueKeyDelegate` or `IEntityNotFoundByPropertyRefDelegate`?

Reply:
I like `IEntityNotFoundByPropertyRefDelegate` better.

Reply:
> IEntityNotFoundByUniqueKeyDelegate or IEntityNotFoundByPropertyRefDelegate?

Any of them.

Reply:
Thank you for reviewing the commit I have made the requested changes please let me know if there is anything else I can do.